README.md

```diff
@@ -4,6 +4,50 @@ My solutions to [Advent of Code 2023].
 
 [Advent of Code 2023]: https://adventofcode.com/2023
 
+## To run
+Use :
+```lein run -m t.day01```
+in the terminal.
+
+Change `day01` as needed.
+
+## Lessons (learned?)
+
+### Day 01
+
+Using `Integer.` is a bad idea.\
+Capital Letter ==> java class\
+Capital Letter + "." ==> constructor of the java class :\
+    - creates new object
+    - Integer : in 32bits  ==> max is 2.10^9 : not that much with advent of code and no warning/error
+Solution : better to use integer 64bit: Long\
+Let's **not** use `Long.` because again new object... instead : **`parse-long`**
+
+
+### Day 02
+
+#### `reduce` and `and`
+
+```clojure
+(defn iscolorok?
+  [color numb cubes]
+  (reduce (fn [a b] (and a b)) true
+          (map (fn [b] (<= (parse-long (re-find #"\d+" b)) numb))
+               (re-seq (re-pattern (str "\\d+ " color)) cubes))))
+```
+`reduce` needs a function of two elements\
+`and` is _not_ a function so we have to define one\
+we set `true` as a starting point in case the sequence has only one element
+
+#### destructuring
+ 
+with `fn [input]`\
+and `input` is `[game color]`\
+we can directly write: `fn[[game color]]` and call `game` and `color` in the rest
```

Comment:
Worth bearing in mind that if you happen to call the function with a vector of three or more elements, the other ones get silently ignored. You can use the `&` notation to capture the "tail" ("rest"/"other") elements, e.g.

```clojure
t.core=> ((fn [[a b & c]] (prn [a b c])) [1 2 3 4 5])
[1 2 (3 4 5)]
nil
t.core=> (let [[a b & c] [1 2 3 4 5]] (prn [a b c]))
[1 2 (3 4 5)]
nil
t.core=>
```

"destructuring" works the same way at every "binding" point, so `fn` args, `defn` args, `let` bindings, `bindings`, `if-let`, `when-let`, etc. It's recursive, too:

```clojure
t.core=> (let [[a [b c] & d] [[1 2] [3 4] [5 6] [7 8]]] {:a a, :b b, :c c, :d d})
{:a [1 2], :b 3, :c 4, :d ([5 6] [7 8])}
t.core=>
```